x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -162,8 +266,28 @@ const ThreatMapping = ({ threatMapping }: ThreatMappingProps) => {
   return <EuiText size="s">{description}</EuiText>;
 };
 
+interface NewTermsFieldsProps {
+  newTermsFields: string[];
+}
+
+const NewTermsFields = ({ newTermsFields }: NewTermsFieldsProps) => (
+  <BadgeList badges={newTermsFields} />
+);
+
+interface HistoryWindowSizeProps {
+  historyWindowStart?: string;
+}
+
+const HistoryWindowSize = ({ historyWindowStart }: HistoryWindowSizeProps) => {
+  const size = historyWindowStart ? convertHistoryStartToSize(historyWindowStart) : '7d';
+
+  return <EuiText size="s">{size}</EuiText>;
+};
+
+// eslint-disable-next-line complexity
 const prepareDefinitionSectionListItems = (
-  rule: Partial<RuleResponse>
+  rule: Partial<RuleResponse>,
```

Comment:
Why is this typed as `Partial<RuleResponse>` when it appears the types that are being passed into this are `RuleResponse`?

Reply:
Well, the initial idea was to make these section components as reusable as possible so that a component user wouldn't have to pass all the rule properties in. It would just render whichever properties are passed. Now as you mentioned it, I'm thinking actually maybe there's no real use case for this.

And also my idea was to avoid `rule.type` checks in the component somehow. But since I ended having a bunch of property existence checks anyways, using Partial doesn't make sense.

I am going to remove the `Partial` and all the extra checks. Thx for suggestion!